client/js/src/cache_map.js

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2014 Treode, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+var tx_clock = require('./tx_clock');
+/**
+*	Constructor
+*/
+
+function cache_map(cache_limit){
+	this._size  = 0;
+	this._limit = cache_limit;
+	this._map   = {};	
+	this._head = null;
+	this._tail = null;
+}
+
+cache_map.prototype = {
+	constructor: cache_map,
+
+	/**
+	*	Description: Adds an object to the head of our list
+	*	@private
+	*/
+	_add:function(node)
+	{
+		if(this._size == 0)			//if this is the only item
+		{
+			this._head = node; 
+			this._tail = node;
+			return 1;
+		}
+		this._head.next = node;
+		node.previous = this._head;
+		this._head = node;
+		return 1;
+	},
+	/**
+	* Description: Converts input params into tx_clocks, based on whether there exists
```

Comment:
We know this is the description; there's no need to reiterate that. That applies generally throughout.

I cannot make heads of tails of this description.